### ISSUE TITLE
--user no longer writes .vmdk in that user's directory

### ISSUE THREAD

**Issue by adriaandegroot** [RainbowHackerHorse/vzvol#32]
On FreeBSD -- so maybe this should be a downstream PR first -- the `.vmdk` files are no (longer, at least in my memory) created in the VBoxdisks directory in the user's home-dir, and are also not given permissions suitable for that user. E.g., on my home machine running 11-STABLE and port version 0.6.1,:

```
[moi@beastie ~]$ sudo vzvol --user moi -s 1G -t virtualbox -p zpool -v example
Testing to ensure zvol was created
Creating /root/VBoxdisks/example.vmdk
RAW host disk access VMDK file /root/VBoxdisks/example.vmdk created successfully.
Please use /root/VBoxdisks/example.vmdk as your VM Disk
[moi@beastie ~]$ ls -la /root/VBoxdisks/example.vmdk
-rw-------  1 root  wheel  551 May 24 15:11 /root/VBoxdisks/example.vmdk
```

This is (partly) because sudo in the default configuration drops $HOME from the environment, so it gets re-initialized to root's home. Can this be made smarter in vzvol? Or (downstream) a pkg-message could be added explaining how to configure sudo to preserve $HOME.

In addition, the permissions are not set on the .vmdk file to allow the user for whom the volume was created (and who has permissions from ZFS for the volume) to read the file in order to use it in VirtualBox.

**Comment by RainbowHackerHorse** [comment]
Ah crap. You're right. I should have caught this. I need to dig further into sudo and see what variables it exposes

**Comment by RainbowHackerHorse** [comment]
Resolved in 0.6.3
